src/ucp/core/ucp_am.c

```diff
@@ -155,22 +155,36 @@ static void ucp_am_rndv_send_ats(ucp_worker_h worker,
                           UCP_AM_ID_RNDV_ATS, "send_ats");
 }
 
+static UCS_F_ALWAYS_INLINE void ucp_am_release_long_desc(ucp_recv_desc_t *desc)
+{
+    /* Don't use UCS_PTR_BYTE_OFFSET here due to coverity false positive report.
+     * Need to step back by am_malloc_offset, where originally allocated pointer
+     * resides. */
+    ucs_free((char*)desc - desc->am_malloc_offset);
+}
+
+static UCS_F_ALWAYS_INLINE int
+ucp_am_rdesc_in_progress(ucp_recv_desc_t *desc, ucs_status_t status)
```

Comment:
status -> am_cb_status
maybe add assertion of invalid combinations of return status and desc flags?